refactor: convert to typescript and hono

kobik opened the pull request.

kobik committed:
feat: add support for gemini model

kobik committed:
refactor: convert to typescript and hono

omer182 commented on src/lib/llm/AnthropicProvider.ts:
```diff
@@ -0,0 +1,138 @@
+      throw new Error("Anthropic API key is required.");
+    }
+    this.anthropic = new Anthropic({ apiKey });
+    this.modelName = 'claude-3-sonnet-20240229'; // Updated to a generally available Sonnet model
```
use `claude-3-7-sonnet-20250219`

kobik replied:
fixed

omer182 commented on src/config/index.ts:
```diff
@@ -0,0 +1,42 @@
+export const profileMap: ProfileMap = {
+  heb: {
+    profileId: 7,
+    // TODO: is this really needed?
```
its not needed. we can deduce that in the search function from the intent.type so we drop these. also the tags aren ot needed. and the language profile. basically all we need here ithe profile id "(heb). so we can just use that in the search depends on the profile. and the rootFolder depends on the media type

kobik replied:
i'll handle that in another MR

omer182 commented:
we also need to address the dockerfile. built it correctly

kobik committed:
build: update Dockerfile for multi-stage build and improve production setup

- Changed base image to Node.js 22-alpine for a smaller footprint.
- Added multi-stage build process to separate dependencies and built app.
- Updated npm install command to npm ci for cleaner installs.
- Enhanced build command in package.json to specify tsconfig.
- Implemented graceful shutdown handling in the server for better resource management.

kobik committed:
chore: remove outdated documentation for generic model call feature

- Deleted design and implementation plan documents for the generic model call support, as they are no longer relevant.
- Updated AnthropicProvider to reflect the latest model version for improved compatibility.

kobik committed:
refactor: make ProfileConfig properties optional

- Updated ProfileConfig interface to have optional properties for better flexibility.
- Removed hardcoded values from profileMap for 'heb' and 'default' profiles, allowing for dynamic configuration.

kobik committed:
chore: configure nodemon and update package dependencies

- Added a new nodemon configuration file to streamline development with TypeScript.
- Replaced `ts-node` with `tsx` in package.json for improved TypeScript execution.
- Updated package-lock.json to reflect new dependencies and removed outdated ones.
- Introduced new VSCode settings for spell checking specific terms related to the project.

kobik committed:
chore: update package dependencies and Gemini model version

- Bumped `axios` to version 1.9.0 and `dotenv` to version 16.5.0 in package.json.
- Updated `@types/node` to version 22.15.24 for improved type definitions.
- Changed the Gemini model from "gemini-1.5-flash-latest" to "gemini-2.5-flash-preview-05-20" in GeminiProvider.ts for enhanced performance.

kobik committed:
chore: add ESLint configuration and update workflows

- Introduced a new ESLint configuration file (.eslintrc.json) to enforce coding standards and improve code quality.
- Updated package.json to include linting and testing scripts.
- Added a new GitHub Actions workflow (pr-check.yml) for pull request checks, including linting and dependency installation.
- Replaced the deprecated Docker workflow (docker-ghcr.yml) with a new build and publish workflow (docker-build.yml) for better Docker image management.

kobik committed:
chore: enhance ESLint configuration and improve error handling

- Updated .eslintrc.json to extend TypeScript ESLint rules and added TypeScript parser for better linting support.
- Modified package.json to include TypeScript ESLint plugin and parser as devDependencies.
- Improved error handling in various service files by changing catch blocks to handle unknown errors more gracefully.
- Refactored error logging to provide clearer messages when exceptions occur.

kobik committed:
chore: update Node.js version in GitHub Actions workflow

- Changed Node.js version from 20 to 22 in the pr-check.yml workflow for improved compatibility and performance.

kobik committed:
chore: update package dependencies and ESLint configuration

- Removed outdated esbuild modules from package-lock.json to streamline dependencies.
- Added eslint back to package.json to ensure consistent linting across the project.

kobik committed:
Merge remote-tracking branch 'upstream/main' into typescript

kobik committed:
chore: remove LLM provider implementations and documentation

- Deleted the AnthropicProvider and GeminiProvider classes, along with their associated files, to streamline the codebase.
- Removed the ModelProvider base class and related factory functions for LLM provider creation.
- Eliminated the README.md file, which contained project overview and setup instructions, as part of the documentation cleanup.
